tests/direct.js

```diff
@@ -121,40 +121,59 @@ import { Tinytest } from 'meteor/tinytest';
 //   })
 // })
 
-[{}, { connection: null }].forEach(function (conntype, i) {
-  [null, 'direct_collection_test_stringid'].forEach(function (ctype) {
-    const cname = ctype && (ctype + i)
-    Tinytest.add(`direct - update and remove should allow removing by _id string (${cname}, ${JSON.stringify(conntype)})`, function (test) {
-      const collection = new Mongo.Collection(cname, conntype)
-
-      // Full permissions on collection
-      collection.allow({
-        insert: function () { return true },
-        update: function () { return true },
-        remove: function () { return true }
-      })
-
-      function hasCountAndTestValue (count, value) {
-        const cursor = collection.direct.find({ _id: 'testid', test: value })
-        test.equal(cursor.count(), count)
-      }
-
-      collection.direct.remove({ _id: 'testid' })
-      collection.direct.insert({ _id: 'testid', test: 1 })
-      hasCountAndTestValue(1, 1)
-      collection.direct.update('testid', { $set: { test: 2 } })
-      hasCountAndTestValue(1, 2)
-      collection.direct.remove('testid')
-      hasCountAndTestValue(0, 2)
+// TODO(v3): failing on client
+// [{}, { connection: null }].forEach(function (conntype, i) {
+//   [null, 'direct_collection_test_stringid'].forEach(function (ctype) {
+//     const cname = ctype && (ctype + i)
+//   })
+// })
+
+function createTest (cname, conntype) {
+  Tinytest.addAsync(`direct - update and remove should allow removing by _id string (${cname}, ${JSON.stringify(conntype)})`, async function (test) {
+    const collection = new Mongo.Collection(cname, conntype)
+
+    // Full permissions on collection
+    collection.allow({
+      insert: function () { return true },
+      update: function () { return true },
+      remove: function () { return true },
+      insertAsync: function () { return true },
+      updateAsync: function () { return true },
+      removeAsync: function () { return true }
     })
+
+    async function hasCountAndTestValue (count, value) {
+      const cursor = await collection.direct.find({ _id: 'testid', test: value })
+      test.equal(await cursor.countAsync(), count)
+    }
+
+    await collection.direct.removeAsync({ _id: 'testid' })
+    await collection.direct.insertAsync({ _id: 'testid', test: 1 })
+
+    // TODO(v3): returns []
+    console.log(await collection.find().fetchAsync())
```

Comment:
For some reason this fails on the client. It appears _id: 'testid' document is never inserted properly.
I tried to remove removeAsync before it but this didn't change anything.

Reply:
Looks like we have to use `resolveType: 'stub'`. https://github.com/meteor/meteor/issues/13036